### DISCUSSION TITLE
Complex terrain corrections

### DISCUSSION THREAD

**Discussion by mdtanker** [fatiando/community#104] [Q&A] [answered]
I was hoping to get some clarification on performing topography corrections of gravity disturbances to attain a topo-free gravity disturbance. As shown in the [Fatiando tutorials](https://www.fatiando.org/tutorials/notebooks/gravity-processing.html#topographic-correction), this is done by forward modeling and removing the gravity effects resulting from any topographic deviations from the normal Earth (the ellipsoid). I take this to include 1) any masses above the ellipsoid which don't equal the density of air ($\rho_{air}$), and 2) any masses below the ellipsoid which don't equal the assumed constant density of the crust ($\rho_{crust}$ ). From the below figure, this typically includes
* Crust above ellipsoid
* Water above ellipsoid
* Air below ellipsoid
* Water below ellipsoid
For these topographic masses, their assigned density is the difference between their true density and the density of air or crust, based on if they are above or below the ellipsoid, respectively.

Question 1: Should the density of prisms above the ellipsoid technically be $\rho_{crust}$ - $\rho_{air}$, but $\rho_{air}$ is omitted since it is close to 0? Or am I incorrect in the above assumptions?

Question 2: The small bit of continent below the ellipsoid should be labeled as air below the ellipsoid correct? And therefore it's density should be  $\rho_{air}$- $\rho_{crust}$. 

Question 3: Similarly, the water above the ellipsoid should have a density of  $\rho_{water}$ - $\rho_{air}$ correct?

<img src="https://github.com/fatiando/community/assets/81199856/eb79b187-fe87-4a94-b496-9a279d446cce" width="500">

Based on these assumptions, I have drafted the below figure to extend the topography correction to ice-covered regions. For this scenario, the anomalous masses above the ellipsoid (red line) include ice, water and earth, and anomalous masses below the ellipsoid include air, ice, and water, with the density shown in d). 

Question 4: Would the total topo-effect be the summation of all the gravity effects of the masses (and associated densities) shown in Figure d). Therefore the topo-corrected gravity disturbance would be the gravity disturbance minus these summed effects? Note that the geoid is very negative for positions of Antarctica (-60m), so in these places, the "air below ellipsoid" (green color) is significant, especially since it has a large negative density of $\rho_{air}$ - $\rho_{crust}$.

<img src="https://github.com/fatiando/community/assets/81199856/168ce317-71f4-42e8-9f2d-28c0e5d9c2d6" width="300">

Using the configuration and densities of Fig d), I have calculated these terrain effects for Antarctica, as shown in the profile below. The second profile shows the disturbance, total topo-effect, and the topo-free disturbance. The profile location is shown in the last plot. 

<img src="https://github.com/fatiando/community/assets/81199856/840a81b2-0db3-4ab0-aa49-4e80130525b3" width="500">
<img src="https://github.com/fatiando/community/assets/81199856/7c0cc870-d544-482b-9f0a-9acba899dc2d" width="500">
<img src="https://github.com/fatiando/community/assets/81199856/b48d6200-ac58-4754-92b5-1e57ef6f21d5" width="500">

I'm confused about the distinction between a topographic correction, and the "stripping" of layers, as used in geodesy (discussed in [Vajda et al. 2008](http://link.springer.com/10.1007/s11200-008-0003-5)). They make distinctions between correcting for the effects of the water itself, and the density contrast between the water and the seafloor. 

Instead of the confusing separation of all the masses above and below the ellipsoid, could I just calculate the gravity effect of each layer, using absolute densities (i.e. the ice layer, from the ice surface to the ice base, with the density of ice) and remove that effect from the disturbance? 

I've had a look at several Vajda papers and [Understanding the Bouguer Anomaly](https://linkinghub.elsevier.com/retrieve/pii/B9780128129135000087), but haven't found a clear source of info on this. 

Any help is much appreciated!

@LL-Geo, any insight here, from your work in Antarctica? 

Related to #80.

**Comment by LL-Geo** [comment, marked as answer]
Hey @mdtanker,

A key concept in the potential field is the disturbance/anomaly. The things we talk about are the differences between our observations and an idealized, homogeneous Earth. The free-air disturbance is the difference between our measurement and that of the normal Earth at the same location. If we want to see the internal density distribution, we should remove all the terrain effects from the free-air disturbance. One approximation method is to remove the mass based on the geoid. This approach is simple for dealing with ocean mass, but it's not rigorously correct. As you showed, the correct way is quite complex, and it becomes even more so with the presence of an ice sheet.

> Question 1: Should the density of prisms above the ellipsoid technically be ρcrust - ρair, but ρair is omitted since it is close to 0? Or am I incorrect in the above assumptions?

Yes, this is correct. The ellipsoid is an idealised approximation of Earth, aka normal Earth. So anything above the ellipsoid surface should have 0 density for the normal Earth.

> Question 2: The small bit of continent below the ellipsoid should be labeled as air below the ellipsoid correct? And therefore it's density should be ρair- ρcrust.

Yes, you are correct. The area of air below the ellipsoid represents the space where air has replaced the crust. 

> Question 3: Similarly, the water above the ellipsoid should have a density of ρwater - ρair correct?

Yup.

> Question 4: Would the total topo-effect be the summation of all the gravity effects of the masses (and associated densities) shown in Figure d). Therefore the topo-corrected gravity disturbance would be the gravity disturbance minus these summed effects? Note that the geoid is very negative for positions of Antarctica (-60m), so in these places, the "air below ellipsoid" (green color) is significant, especially since it has a large negative density of ρair - ρcrust.

That's a nice figure!  Yes, we need to consider the geoid variation for the concept of free-air disturbance and the terrain effect. One further complicity is that the "terrain density" could vary in locations, especially for the continental scale model. 

**One easy way to think about the problem is that we can have two types of mass. One is mass above topography/bathymetry, and the other is mass between topography and the ellipsoid. So what we need is to remove the mass above the topography and the mass/deficit between the topography and the ellipsoid. Same for your Fig (d), we only need to - Pearth, when topography/bathymetry is below ellipsoid.** 

So for case 1: Just topography: we have topography effect = mass above topography (0) +  mass extra between topography and the ellipsoid

And for a more complex case: Near the groundline: The topography effect = mass above bathymetry ( ice + water+air ) + mass deficit between bathymetry and the ellipsoid ( -pCrust with the thickness between bathymetry and the ellipsoid, the mass deficit is negative) 


> I'm confused about the distinction between a topographic correction, and the "stripping" of layers, as used in geodesy (discussed in [Vajda et al. 2008](http://link.springer.com/10.1007/s11200-008-0003-5)). They make distinctions between correcting for the effects of the water itself, and the density contrast between the water and the seafloor.

Yes, I'm a bit confused about this paper as well. I think the terminology is different in their paper, the gNET only remove the completed topography effect in the onshore and offshore areas where the bathymetry is above the ellipsoid. For area bathymetry below the ellipsoid, it replaces air with water.  And for the next step, they use "stripping" to replace water with crust to get the BT.

 If we combine them together:

Note: N is negative, because geoid is below ellipsoid.
![image](https://github.com/fatiando/community/assets/54405391/6aaa607a-c2c7-4938-bf14-2c5f9c1db23f)+
![image](https://github.com/fatiando/community/assets/54405391/e7385af6-d1e5-4050-aad7-d63c71139fb3)
= - Distance (Ellipsoid to Sea-level) * p Sea + Distance (Bathemetry to Ellipsoid) * (p Sea - p Crust)
= Distance ( Bathemetry  to Sea-level) * p Sea - Distance (Bathemetry to  Ellipsoid) p Crust
If we separate the last term Distance (Bathemetry to Ellipsoid)  = Distance ( Bathemetry  to Sea-level) + Distance (Ellipsoid to Sea-level)
So we have: 
=Distance ( Bathemetry  to Sea-level) * p Sea - Distance ( Bathemetry  to Sea-level) * p Crust - Distance (Ellipsoid to Sea-level) * p Crust
=Distance ( Bathemetry  to Sea-level) *  (p Sea - p Crust) + Distance (Ellipsoid to Sea-level) * (p Air - p Crust)  


> Instead of the confusing separation of all the masses above and below the ellipsoid, could I just calculate the gravity effect of each layer, using absolute densities (i.e. the ice layer, from the ice surface to the ice base, with the density of ice) and remove that effect from the disturbance?
 
If you just use the absolute value, you are missing some mass here. Removing the gravity effect using absolute value is the first step. You need also to consider the mass between the ellipsoid and topography. See the paragraph in bold.

Some modelling work uses free-air, and the way they handle the topography is by using the absolute density value for each layer (but the model should influence crust and the model bottom should be deeper enough (at least deeper than the deepest point in bathymetry). And the model will remove the mean value of the computed gravity effect. The mean value of the computed gravity effect could refer to step 2, removing the mass between topography/bathymetry with ellipsoid. 

> @LL-Geo, any insight here, from your work in Antarctica?

Nicely done for the Antarctica stuff, but one thing to bear in mind is ensuring that the gravity and topography information have similar resolutions. As you can see, the satellite free-air data has a lower resolution than the terrain effect data, which leads to the Bouguer correction being over-corrected for small-scale terrain effects.

**Comment by santisoler** [comment]
Wow! There's a lot going on this, and those images are awesome.

Just wanted to mention something important when working with satellite data. I tend to avoid downloading the gravity **on the ellipsoid**. I choose a certain height above it, usually something higher than the highest topography point, so all observation points are outside the masses.

What @LL-Geo mentions about the resolution between the gravity data and the topography is true: the gravity data has much longer wavelengths than the topography effect. This usually suggest that the gravity data is filtered up to certain wavelength, and because this is satellite data we know that for sure. But I'm more concern about the difference in scales between them. Seeing that makes me think that there might be an issue with the height of the observation points.

At which height do you have your observation points? Are you computing the terrain effect using those heights?

**Comment by mdtanker** [comment]
Thanks for the detailed response @LL-Geo! That clarifies some of my thoughts. I agree with the resolution issues of the Antarctic-wide correction. I've resampled the topo grids at a lower resolution, which help reduce the short wavelength component of the topography effect. 

As @santisoler pointed out, the difference in scales between the disturbance and the terrain effect is really interesting. I checked and I am using the correct observation heights for both calculating the disturbance and for the terrain effect.  The gravity data I was using here was from [EIGEN-6C4](https://www.fatiando.org/ensaio/latest/api/generated/ensaio.fetch_earth_gravity.html#ensaio.fetch_earth_gravity), so to compute both the normal gravity and the terrain effect I used geometric observation heights of 10 km.

The ratio between the magnitudes of the disturbance and the terrain effect here, ~1:2, (~250mGal vs ~500mGal, respectively), actually seem comparable to calculations for other planets. The two below figures are from [Tenzer et al. 2019](http://link.springer.com/10.1007/s10712-018-9504-0) and show the range in disturbances for Mercury, Venus, Earth and the Moon are all approximately half the range of their respective terrain effects.  

<img src=https://github.com/fatiando/community/assets/81199856/b942e97d-ee2e-4c79-a65b-41ebf04bcfc1 width="425"/> <img src=https://github.com/fatiando/community/assets/81199856/8f16f6fa-9fab-41b0-97a0-96bfe1e87be8 width="425"/> 

Also, I assume applying an isostatic correction would partially flatten the topo-correction, reducing its overall magnitude. Should we try and synthesize all of this into a slightly more complete tutorial in the documentation at some point? Antarctica might be a good example for explaining topo-corrections since it's one of the more complex scenarios. Or maybe to explain the process clearly a more simple scenario would be better.

**Reply by LL-Geo** [reply]
That would be a great idea @mdtanker 👍

Just would like to point out, that although [EIGEN-6C4](https://www.fatiando.org/ensaio/latest/api/generated/ensaio.fetch_earth_gravity.html#ensaio.fetch_earth_gravity) has globally resolution up to 10 km (2190 spherical harmonic). The actual resolution in Antarctica is not higher enough, which is mainly due to the limited airborne and ground-based data in the model. The main source in EIGEN-6C4 is GRACE and GOCE data, which are up to (235 spherical harmonic degrees), which give spatial resolution up to 90 km. See the [ANTGG](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1002/2015GL067439) paper.

**Reply by mdtanker** [reply]
That's a good point, thanks! Are there any higher-res alternatives you know of for Antarctica? The updated [AntGG](https://ftp.space.dtu.dk/pub/RF/4D-ANTARCTICA/4D_antarctica_gravity-grid.pdf) looks good, but doesn't actually provide any observation heights for any of the data :unamused:.
